src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1005,8 +1029,18 @@ def get_completer_info(self, prefix="", excluded=None) -> List[List[str]]:
 class Command(Action):
     """Command object."""
 
-    def __call__(self, **kwds):
-        """Call a query with the specified keyword arguments."""
+    def __call__(self, purpose: Optional[str] = None, **kwds):
+        """Call a command with the specified keyword arguments."""
+        file_purpose = None
+        if hasattr(self, "file_name"):
+            file_purpose = self.file_name.get_attr(_InlineConstants.file_purpose)
+        elif hasattr(self, "file_name_list"):
+            file_purpose = self.file_name_list.get_attr(_InlineConstants.file_purpose)
+        elif hasattr(self, "filename"):
+            file_purpose = self.filename.get_attr(_InlineConstants.file_purpose)
+        file_purpose = purpose if purpose else file_purpose
```

Comment:
We are computing `file_purpose` and then only using it if `purpose` is not set. Please consider this point. Please also extract all this functionality into a separate method. This is complicting the implementation of `__call__` .

Reply:
@seanpearsonuk Yes this is old code. We have extracted file_purpose in settings static info. Now we are working for meshing datamodel static info. We will update this code soon.

Reply:
`purpose` is temporary for testing in Ansys Lab. We will remove it soon. We have moved this functionality into a separate method.